test/e2e/backups/backups.go

```diff
@@ -0,0 +1,190 @@
+
+		// refresh cluster
+		Expect(f.Client.Get(context.TODO(), clusterKey, cluster)).To(Succeed(),
+			"failed to get cluster %s", cluster.Name)
+
+		// write test data to the cluster
+		testData = f.WriteSQLTest(cluster, 0, rootPwd)
+	})
+
+	Describe("tests with a good backup", func() {
+		var (
+			backup *api.MysqlBackup
+		)
+		BeforeEach(func() {
+			By("create a backup for cluster")
+			backup = framework.NewBackup(cluster, bucketName)
+			Expect(f.Client.Create(context.TODO(), backup)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			// delete the backup that was created
+			f.Client.Delete(context.TODO(), backup)
+		})
+
+		It("should be successful and can be restored", func() {
+			// check that the data is in the cluster
+			actual := f.ReadSQLTest(cluster, 0, rootPwd)
+			Expect(actual).To(Equal(testData))
+
+			By("check backup was successful")
+			// should be backup successfully
+			Eventually(f.RefreshBackupFn(backup), timeout, POLLING).Should(
+				framework.BackupCompleted())
+			Eventually(f.RefreshBackupFn(backup), timeout, POLLING).Should(
+				framework.HaveBackupCond(api.BackupComplete, core.ConditionTrue))
+
+			backup = f.RefreshBackupFn(backup)()
+
+			By("create a new cluster from init bucket")
+			// create cluster secret
+			name := fmt.Sprintf("cl-%d-2", rand.Int31()/1000)
+			sct := framework.NewClusterSecret(name, f.Namespace.Name, rootPwd)
+			Expect(f.Client.Create(context.TODO(), sct)).To(Succeed(), "create cluster secret")
+
+			// create cluster
+			cl := framework.NewCluster(name, f.Namespace.Name)
+			cl.Spec.InitBucketSecretName = backupSecret.Name
+			cl.Spec.InitBucketURI = backup.Spec.BackupURL
+			Expect(f.Client.Create(context.TODO(), cl)).To(Succeed(),
+				"failed to create cluster '%s'", cluster.Name)
+
+			// wait for cluster to be ready
+			Eventually(f.RefreshClusterFn(cl), f.Timeout, POLLING).Should(
+				framework.HaveClusterReplicas(1))
+			Eventually(f.RefreshClusterFn(cl), f.Timeout, POLLING).Should(
+				framework.HaveClusterCond(api.ClusterConditionReady, core.ConditionTrue))
+
+			// check the data that was read before
+			actual = f.ReadSQLTest(cl, 0, rootPwd)
+			Expect(actual).To(Equal(testData))
+
+		})
+	})
+
+	It("should failed the backup if bucket does not exists", func() {
+		backup := framework.NewBackup(cluster, "gs://does_not_exist")
+		Expect(f.Client.Create(context.TODO(), backup)).To(Succeed())
+
+		localTimeout := 150 * time.Second
+		// checks for the job because the backup is updated after the job is
+		// marked as failed
+		Eventually(func() *batchv1.Job {
+			j := &batchv1.Job{}
+			key := types.NamespacedName{
+				Name:      framework.GetNameForJob(backup),
+				Namespace: backup.Namespace,
+			}
+			f.Client.Get(context.TODO(), key, j)
+			return j
+		}, localTimeout, POLLING).Should(WithTransform(
+			func(j *batchv1.Job) int32 { return j.Status.Failed }, BeNumerically(">", 2)))
+	})
+
+	It("should take the backup from replica", func() {
```

Comment:
This should be a unit test and we should tests backups/restores with 2 node clusters.